ios/Tests/MappersTests.swift

```diff
@@ -19,4 +19,184 @@ final class MappersTests: XCTestCase {
+            ToggleResult.enabled.rawValue as NSNumber,
+            ToggleResult.skipped.rawValue as NSNumber,
+        ])
+        let selectionResult = TestableSelectionResult(skipped: false, selection: "Yes", toggles: [
+            ToggleResult.enabled.rawValue as NSNumber,
+            ToggleResult.skipped.rawValue as NSNumber,
+        ])
+        let signatureResult = TestableSignatureResult(skipped: false, signatureSvg: "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 974 943\" fill=\"none\" stroke-linecap=\"round\" stroke-linejoin=\"round\" stroke-width=\"10\" stroke=\"black\"><g><path d=\"M468.5171463.52472 L468.5171 463.52472 \"/></g></svg>", toggles: [
+            ToggleResult.enabled.rawValue as NSNumber,
+            ToggleResult.skipped.rawValue as NSNumber,
+        ])
+        var output: NSDictionary = Mappers.mapFromCollectInputsResults([
+            textResult
+        ])
+        XCTAssertNotNil(output.object(forKey: "collectInputResults"))
+        XCTAssertTrue(output["collectInputResults"] is [NSDictionary])
+
+        var results: [NSDictionary] = output["collectInputResults"] as! [NSDictionary]
+
+        XCTAssertEqual(results.count, 1)
+        guard let result = results.first else {
+            XCTFail("CollectInput result should have had a result")
+            return
+        }
+        guard
+            let skipped = result["skipped"] as? Bool,
+            let toggles = result["toggles"] as? [String],
+            let text = result["text"] as? String else {
+            XCTFail("CollectInput TestResult should have had text, skipped and toggles")
+            return
+        }
+        XCTAssertFalse(skipped)
+        XCTAssertEqual(toggles.count, 2)
+        XCTAssertEqual(toggles[0], "enabled")
+        XCTAssertEqual(toggles[1], "skipped")
+        XCTAssertEqual(text, "Written text from the reader")
+        
+        output = Mappers.mapFromCollectInputsResults([
+            numericResult, phoneResult, emailResult, selectionResult, signatureResult
+        ])
+        XCTAssertNotNil(output.object(forKey: "collectInputResults"))
+        XCTAssertTrue(output["collectInputResults"] is [NSDictionary])
+        
+        results = output["collectInputResults"] as! [NSDictionary]
+        
+        let testNumericString = "numericString"
+        let testPhone = "phone"
+        let testEmail = "email"
+        let testSelection = "selection"
+        let testSignatureSvg = "signatureSvg"
+        XCTAssertEqual(results.count, 5)
+        XCTAssertTrue(results.contains(where: { $0[testNumericString] != nil }))
+        XCTAssertTrue(results.contains(where: { $0[testPhone] != nil }))
+        XCTAssertTrue(results.contains(where: { $0[testEmail] != nil }))
+        XCTAssertTrue(results.contains(where: { $0[testSelection] != nil }))
+        XCTAssertTrue(results.contains(where: { $0[testSignatureSvg] != nil }))
```

Comment:
Can we update these checks to assert the order of each result is kept? I think we can loop through the array naively or even hardcode (eg, `results[0][testNumericString] != nil`)

Reply:
ok, let me improve that.